internal/dcache/cluster_manager/cluster_manager.go

```diff
@@ -3140,30 +3179,54 @@ func (cmi *ClusterManager) batchUpdateComponentRVState(msgBatch []*dcache.Compon
 				continue
 			}
 
+			//
+			// If the RV/MV combination is already present in the batch, we can ignore this update as it
+			// was already processed in the previous iteration. We do assert that the previous state
+			// is same as the new state requested, so that we don't end up with multiple updates for the same
+			// RV/MV combination in the batch.
+			//
+			if rvPrevState, ok := existing[mvName+rvName]; ok {
+				_ = rvPrevState
+				common.Assert(rvPrevState == rvNewState)
+				log.Debug("ClusterManager::batchUpdateComponentRVState: %s/%s, ignoring duplicate state change (%s -> %s)",
+					rvName, mvName, currentState, rvNewState)
+				ignoredCount++
+				continue
+			}
+
 			//
 			// and the new state requested must be valid.
 			// Note that we support only few distinct state transitions.
 			//
 			if currentState == dcache.StateOutOfSync && rvNewState == dcache.StateSyncing ||
 				currentState == dcache.StateSyncing && rvNewState == dcache.StateOnline ||
 				currentState == dcache.StateSyncing && rvNewState == dcache.StateOutOfSync ||
-				currentState == dcache.StateSyncing && rvNewState == dcache.StateOffline ||
-				currentState == dcache.StateOnline && rvNewState == dcache.StateOffline {
+				currentState == dcache.StateSyncing && rvNewState == dcache.StateInbandOffline ||
+				currentState == dcache.StateOnline && rvNewState == dcache.StateInbandOffline {
 
 				log.Debug("ClusterManager::batchUpdateComponentRVState: %s/%s, state change (%s -> %s)",
 					rvName, mvName, currentState, rvNewState)
 				successCount++
 			} else {
 				//
 				// Following transitions are reported when an inband PutChunk failure suggests an RV as offline.
-				// StateOnline  -> StateOffline
-				// StateSyncing -> StateOffline
+				// StateOnline  -> StateInbandOffline
+				// StateSyncing -> StateInbandOffline
 				//
 				// Since we can have multiple PutChunk requests outstanding, all but the first one will find the
-				// currentState as StateOffline, we need to ignore such update requests.
+				// currentState as StateInbandOffline, we need to ignore such update requests.
+				//
+				// We can also get this transition,
+				// StateOffline -> StateInbandOffline
+				// We can get this when the RV is offline and its status has been updated in the clustermap.
+				// However in inband PutChunk failure, we send to update the RV state to InbandOffline.
+				// Since, the RV is already marked offline in the clustermap, we ignore this request.
```

Comment:
since currentState is the component RV state and not the state of the RV in the RV map, is it possible?